api/src/nlp/services/nlp-entity.service.ts

```diff
@@ -97,4 +125,49 @@ export class NlpEntityService extends BaseService<
     );
     return Promise.all(findOrCreate);
   }
+
+  /**
+   * Clears the NLP map cache
+   */
+  async clearCache() {
+    await this.cacheManager.del(NLP_MAP_CACHE_KEY);
+  }
+
+  /**
+   * Event handler for Nlp Entity updates. Listens to 'hook:nlpEntity:*' events
+   * and invalidates the cache for nlp entities when triggered.
+   */
+  @OnEvent('hook:nlpEntity:*')
+  async handleNlpEntityUpdateEvent() {
+    this.clearCache();
```

Comment:
```suggestion
    await this.clearCache();
```
we can also wrap it into a try catch block